Feat: support .bit resolution icon of domian

JeffreyMa597 opened the pull request:
What problem does this PR solve?
------

As title.
.bit is a did service running on CKB which is similar to [ENS](https://ens.domains/) but much more powerful.
Now we plan to support displaying .bit domain strings and some other functions on our account page.

Figma: 
https://www.figma.com/file/6XNoimRDbFTTNm016rbIdU/Magickbase?node-id=2319%3A17211&t=mT2Em2jeCw95irkR-0
https://www.figma.com/file/6XNoimRDbFTTNm016rbIdU/Magickbase?node-id=721%3A10983&t=CvOGbSPJZCWax7ZC-0

Ref: 
https://github.com/Magickbase/godwoken_explorer/issues/1170
https://github.com/Magickbase/visual-style-guide/issues/15

Check List
------
1、If an address has a .bit domain
    show the .bit domain prior to the address
    show the .bit domain on `Baseinfo` of the account page

| page | view |
| --- | --- |
| Homepage | <img width="600" alt="image" src="https://user-images.githubusercontent.com/22511289/209628786-d99e88bb-5530-49ce-b229-bb3324d86960.png"> |
| Transaction info | <img width="600" alt="image" src="https://user-images.githubusercontent.com/22511289/209633033-4e356f9c-3a15-45ef-a0d6-1b9cb5e8edb7.png"> |
| Account detail | <img width="600" alt="image" src="https://user-images.githubusercontent.com/22511289/209638145-3c49bc06-ef4b-48e1-9e33-06406e3a8706.png"> |
| Bridge tokens | <img width="600" alt="image" src="https://user-images.githubusercontent.com/22511289/209638422-0f471cdd-ca52-495e-a342-69a0c4254e36.png"> |
| Bridge token detail| <img width="600" alt="image" src="https://user-images.githubusercontent.com/22511289/209638639-4eb97b61-3d65-412e-af3b-66cba2865d9f.png">|
| Layer2 tokens | <img width="600" alt="image" src="https://user-images.githubusercontent.com/22511289/209639173-733e125d-6aa6-495b-bc1e-c04b9b290ec3.png"> |
| NFT collections/Multi-token-collections | <img width="600" alt="image" src="https://user-images.githubusercontent.com/22511289/209639558-40e38425-48c9-4371-8b16-8558525eed63.png"> |
| NFT collection/Multi-token-collection | <img width="600" alt="image" src="https://user-images.githubusercontent.com/22511289/209640933-f837fd9c-3f32-4c36-b862-50f93134a8f9.png"> |

2、TODO(waiting for double-check by endpoint.):
If a .bit domain is found by searching, gwscan will navigate to the account page directly, which is different from etherscan.io, on etherscan there's a [ens lookup` page](https://etherscan.io/enslookup-search?search=vitalik.eth), we'll not add a page.


### Test
e2e Test
### Task
none

Keith-CY commented:
![image](https://user-images.githubusercontent.com/7271329/208609016-2b0cb27f-967c-4607-9759-44c9d9254686.png)
The .bit domain is `magickbase.bit` but the address shown is `magickba...base.bit`

FrederLu commented:
1、There is no problem with the maximum width of the outer frame, but the width needs to be changed according to the length of the middle text.
<img width="192" alt="image" src="https://user-images.githubusercontent.com/25676576/208835298-c211e8a0-e475-4034-a9c2-3831d9b7e176.png">

2、The .bit logo in the left-aligned data in the list seems to be a bit larger than the UI draft.
<img width="725" alt="image" src="https://user-images.githubusercontent.com/25676576/208837119-2ad409c6-1d81-4180-83e4-66a0e891c72c.png">
<img width="348" alt="image" src="https://user-images.githubusercontent.com/25676576/208837318-3ab6d972-e319-44b4-82c7-d847eb43e954.png">

Keith-CY commented on components/AccountOverview/index.tsx:
```diff
@@ -284,7 +288,11 @@ const AccountOverview: React.FC<AccountOverviewProps & { refetch: () => Promise<
+        <Polyjuice
+          script={account.script as PolyjuiceCreator['script']}
+          scriptHash={account.script_hash}
+          domain={domain}
```
`PolyjuiceCreator` won't have a .bit domain because it doesn't have an eth address

Keith-CY commented on components/TruncatedAddress.tsx:
```diff
@@ -10,13 +11,17 @@ const TruncatedAddress = ({
   sx = {},
   type,
   monoFont = true,
+  showDomain = false,
```
Why set `showDomain` default `false`, domain will be displayed in most places because it's designed to replace an address in the presentation layer

JeffreyMa597 replied:
@Keith-CY Okay, 
I got it. I'll replace all the places if there exists a domain.

Keith-CY commented on components/TxList/index.tsx:
```diff
@@ -212,10 +216,15 @@ const TxList: React.FC<TxListProps & { maxCount?: string; pageSize?: number }> =
+                    <Address
+                      address={from}
+                      type={item.from_account?.type}
+                      domain={from_alias}
```
Seems that only the transaction list adopts `domain`, but it's expected to be displayed all over the page, except the title of an account page.

On account page, the domain will be displayed next to the address instead of replacing it.

JeffreyMa597 commented:
> 3、The avatar is changed in .bit, but the unmodified avatar is still used on the page.

@FrederLu This feature will be implemented later in another PR.

WhiteMinds commented on components/AccountOverview/index.tsx:
```diff
@@ -267,10 +268,13 @@ const AccountOverview: React.FC<AccountOverviewProps & { refetch: () => Promise<
 
   const getInfoBlock = account => {
     const { type } = account
+    const domain = account?.bit_alias
```
Not necessary `?.`, If the account is null, the previous line of code will throw error

FrederLu commented:
There is no .bit alias for the address in the contract, but abnormal alias information is displayed on the home page.
<img width="1235" alt="image" src="https://user-images.githubusercontent.com/25676576/209525532-b6bb6107-d80d-477e-b964-ddf6ccf29b08.png">

<img width="703" alt="image" src="https://user-images.githubusercontent.com/25676576/209524466-f1e8c75c-d6bc-46d7-be08-2817f14bb6c9.png">

JeffreyMa597 commented:
> There is no .bit alias for the address in the contract, but abnormal alias information is displayed on the home page.
> 
> <img alt="image" width="677" src="https://user-images.githubusercontent.com/25676576/209526360-14b427ad-267b-444b-b287-858bb59113e1.png">
> 
> <img alt="image" width="1215" src="https://user-images.githubusercontent.com/25676576/209526486-6b57ea15-8634-493b-8206-b4740fa11363.png">

@FrederLu @zmcNotafraid  It may be a broken data?
<img width="1413" alt="image" src="https://user-images.githubusercontent.com/22511289/209622077-3e625dbe-db91-410e-8a6f-524ec0aabdbb.png">

Keith-CY commented on pages/multi-token-collections/index.tsx:
```diff
@@ -138,13 +142,11 @@ const MultiTokenCollectionList = () => {
-                            href={`/account/${item.account.eth_address}`}
-                          />
-                        </span>
+                        {domain ? (
+                          <Address address={item.account.eth_address} domain={domain} />
+                        ) : (
+                          <HashLink label={item.account.eth_address} href={`/account/${item.account.eth_address}`} />
+                        )}
```
There were two addresses for pc layout and mobile layout, respectively.
Now there's only one, does it work for mobile layout?

JeffreyMa597 replied:
Fixed.

FrederLu commented:
> > There is no .bit alias for the address in the contract, but abnormal alias information is displayed on the home page.
> 
> @FrederLu @zmcNotafraid It may be a broken data? <img alt="image" width="1413" src="https://user-images.githubusercontent.com/22511289/209622077-3e625dbe-db91-410e-8a6f-524ec0aabdbb.png">

@JeffreyMa597 
After looking at the same interface, `bit_alias` is null, it should be that no alias is set. Compare the account information with the alias set.
<img width="1329" alt="image" src="https://user-images.githubusercontent.com/25676576/209633316-eb5cd431-0adb-4ba6-9c8e-7ee6daf3f00c.png">
<img width="1269" alt="image" src="https://user-images.githubusercontent.com/25676576/209633011-5cf039df-e252-4b6b-b7d0-dc4d453c27c1.png">
<img width="1374" alt="image" src="https://user-images.githubusercontent.com/25676576/209633504-4981089a-8000-415e-9614-38f0ec2e9b9e.png">

FrederLu commented:
In the transaction list, both from/to are `0xcc0af0af911dd40853b8c8dfee90b32f8d1ecad6`, but to does not display the .bit alias.
<img width="1286" alt="image" src="https://user-images.githubusercontent.com/25676576/209639943-8e0340b1-41c0-413a-bc74-fbcc619b6c5e.png">

FrederLu commented:
1、The erc20transfers transmission in Transaction Info does not display the .bit alias.
https://godwoken-explorer-6rmtnt4n4-magickbase.vercel.app/zh-CN/tx/0x2730ea6062e59bb4531ba427258bc8a9a01ea51ad784f28d67617066fb746de2?tab=erc20Records
<img width="1250" alt="image" src="https://user-images.githubusercontent.com/25676576/209795234-f6bd14df-d4fe-49ca-b5f5-62ad60d9b8ab.png">
<img width="1302" alt="image" src="https://user-images.githubusercontent.com/25676576/209795320-a4fd5adf-1c02-4265-a8a3-bf19f7979d19.png">

Keith-CY commented on components/MultiTokenActivityList/index.tsx:
```diff
@@ -165,8 +169,10 @@ const ActivityList: React.FC<
-                  <td>{<Address address={item.from_address} type={item.from_account?.type} />}</td>
-                  <td>{<Address address={item.to_address} type={item.to_account?.type} />}</td>
+                  <td>
+                    {<Address address={item.from_address} type={item.from_account?.type} domain={from_bit_alias} />}
```
Maybe brace is unnecessary

Keith-CY commented:
How is it going?

JeffreyMa597 commented:
> How is it going?

I‘ll update the code a moment later based on the latest update from our endpoint.

JeffreyMa597 commented:
> @JeffreyMa597 1、The erc20transfers transmission in Transaction Info does not display the .bit alias. https://godwoken-explorer-6rmtnt4n4-magickbase.vercel.app/zh-CN/tx/0x2730ea6062e59bb4531ba427258bc8a9a01ea51ad784f28d67617066fb746de2?tab=erc20Records

Fixed.
<img width="1306" alt="image" src="https://user-images.githubusercontent.com/22511289/210408882-a678ea66-15d1-4d22-8e79-babd117e6739.png">

> 2、After the .bit alias is set for the Deployer address, the alias information is not displayed. https://godwoken-explorer-6rmtnt4n4-magickbase.vercel.app/account/0xf8f6173ae4a352de056e278ebc940af972b71de9 

Fixed.
<img width="1305" alt="image" src="https://user-images.githubusercontent.com/22511289/210409961-1755c10a-e7ff-48c0-818f-ad734e80ef4a.png">

Keith-CY commented:
Test case should be updated https://github.com/Magickbase/godwoken-explorer-ui/actions/runs/3839121628/jobs/6537429658#step:6:485 @JeffreyMa597
